src/components/dapp-staking/modals/RegisterDappGeneral.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <div>
+    <div class="tw-mb-6">
+      <label
+        class="
+          tw-block tw-text-sm tw-font-medium tw-text-gray-500
+          dark:tw-text-darkGray-400
+          tw-mb-2
+        "
+        >{{ $t('dappStore.modals.logo') }}</label
+      >
+
+      <input-file :file="imageFromFile" :extension="fileExtensions" @dropFile="onDropFile">
+        <Avatar v-if="!!data.iconFile" :url="data.iconFile" class="tw-mx-auto tw-w-20 tw-h-20" />
+        <icon-base
+          v-else
+          class="tw-h-20 tw-w-20 tw-mx-auto dark:tw-text-darkGray-100"
+          viewBox="0 0 20 20"
+          fill="none"
+          stroke="currentColor"
+          aria-hidden="true"
+        >
+          <icon-document />
+        </icon-base>
+      </input-file>
+    </div>
+
+    <q-input
+      v-model="data.name"
+      outlined
+      label="Name"
+      maxlength="200"
+      class="tw-my-2"
+      :rules="[(v) => (v && v.length > 0) || 'dApp name is required.']"
+    />
+    <q-input
+      v-model="data.address"
+      outlined
+      maxlength="42"
+      label="Contract address"
+      class="tw-my-2"
+      :rules="[(v) => isEthereumAddress(v) || 'Enter a valid EVM contract address.']"
```

Comment:
Although at the moment we only support EVM, it might be a good idea to expand the code check for ss58 address too so that we can easily expand it when we support ink

Reply:
Good point. I will do that.